### ISSUE TITLE
Accessing environment variables via resource binding

### ISSUE THREAD

**Issue by Fatimastm** [karmada-io/karmada#3914]
**Please provide an in-depth description of the question you have**:

I aim to define environment variables within the YAML file of a service and then use it through resource binding:

> 
> apiVersion: policy.karmada.io/v1alpha1
> kind: PropagationPolicy
> metadata:
>   name: pp1 
> spec:
>   resourceSelectors:
>     - apiVersion: apps/v1
>       kind: Deployment
>       name: nginx
>   placement:
>     clusterAffinity:
>       clusterNames:
>         - member1
>         - member2
>         - member3
>     spreadConstraints:
>       - spreadByField: cluster
>         maxGroups: 1
> ---
> apiVersion: apps/v1
> kind: Deployment
> metadata:
>   name: nginx
> spec:
>   selector:
>     matchLabels:
>       app: nginx
>   replicas: 1
>   template:
>     metadata:
>       labels:
>         app: nginx
>     spec:
>       containers:
>       - name: nginx
>         image: nginx:1.14.2
**>         env:
>         - name: time
>           value: "2ns"**
> 

But from my understanding, the resource binding does not encompass this particular information. 
Which optional variables can be introduced to a service or propagation policy to enable their retrieval through resource binding?

**Environment**:
- Karmada version: 1.2.1

**Comment by RainbowMango** [comment]
@Fatimastm Long time no see, I'm glad to see you again. hope you doing well. 

Let me try to understand your question.
> I aim to define environment variables within the YAML file of a service and then use it through resource binding

You defined some environments in a `Deployment`, during the propagation process, a `ResourceBinding` would be created to bind the `Deployment`, and you want to see the `environments` in `ResourceBinding`, right?

**Comment by Fatimastm** [comment]
> @Fatimastm Long time no see, I'm glad to see you again. hope you doing well.
>
@RainbowMango Glad to see you after a while. I've been doing well, thank you. 

> Let me try to understand your question.
> 
> > I aim to define environment variables within the YAML file of a service and then use it through resource binding
> 
> You defined some environments in a `Deployment`, during the propagation process, a `ResourceBinding` would be created to bind the `Deployment`, and you want to see the `environments` in `ResourceBinding`, right?

Yes, That's correct. We would like to add an optional variable in the Deployment of service and use it during scheduling in our plugin, similar to the resource request that can be defined in deployment and are accessible thorugh resource binding.

**Comment by RainbowMango** [comment]
I got your question but I don't get your use case well. I'm not sure why your scheduling plugin needs the environments of a Deployment, can you explain more?

**Comment by Fatimastm** [comment]
> I got your question but I don't get your use case well. I'm not sure why your scheduling plugin needs the environments of a Deployment, can you explain more?

It makes decisions based on additional metrics beyond the resource request standard ones such as CPU, memory, and storage. For instance a scenario where a pod requires minimum bandwidth between two nodes within a cluster. Let me rephrase this more broadly: Is there any flexibility within the propagation policy or deployment parameters to accommodate optional metrics obtained from resource binding? or can resource binding be expanded or extended?

**Comment by RainbowMango** [comment]
> For instance a scenario where a pod requires minimum bandwidth between two nodes within a cluster.

Do you mean how much bandwidth required by a pod is declared by the `env`? And your scheduler plugin would make decisions with it. For instance, if a pod requires more bandwidth, the scheduler would schedule it to a cluster that has sufficient bandwidth. 

> Is there any flexibility within the propagation policy or deployment parameters to accommodate optional metrics obtained from resource binding? or can resource binding be expanded or extended?

Unfortunately, as far as know, currently, no such mechanism to achieve that. 
I'm not sure if it is a normal or typical use case, we lack evidence to extend the [ResourceBinding API](https://github.com/karmada-io/karmada/blob/ddae25cadecd53bbae040abb80728040f6606679/pkg/apis/work/v1alpha2/binding_types.go#L55-L123).

**Comment by Fatimastm** [comment]
> Do you mean how much bandwidth required by a pod is declared by the `env`? And your scheduler plugin would make decisions with it. For instance, if a pod requires more bandwidth, the scheduler would schedule it to a cluster that has sufficient bandwidth.

Yes! 
I also have another question, though I'm not sure if I should ask it here or open a separate issue. 
Is there a queuing mechanism in place for scheduling when multiple services are submitted concurrently? 
The information mentioned here [:](https://karmada.io/docs/developers/customize-karmada-scheduler/#configure-multiple-schedulers) suggests that multiple schedulers can operate simultaneously, but does that imply simultaneous processing of workloads?

**Comment by RainbowMango** [comment]
> The information mentioned [here](https://karmada.io/docs/developers/customize-karmada-scheduler/#configure-multiple-schedulers) suggests that multiple schedulers can operate simultaneously, but does that imply simultaneous processing of workloads?

Nope. Multiple schedulers could work together, but they process(schedule) different workloads, in other words, each workload can only be processed by a scheduler.  For instance, once you declare the [schedulerName in PropagationPolicy](https://github.com/karmada-io/karmada/blob/6dea12190e9752eef937c2461e0223a5fe88ee6a/pkg/apis/policy/v1alpha1/propagation_types.go#L115-L120), all matched workloads would be process by that scheduler.

**Comment by RainbowMango** [comment]
Hi @Fatimastm I wonder are you using Karmada in production now?

**Comment by Fatimastm** [comment]
> Hi @Fatimastm I wonder are you using Karmada in production now?

Hi @RainbowMango.  No, we are using Karmada in research project.
